src/v/s3/client.cc

```diff
@@ -339,6 +341,22 @@ request_creator::make_delete_object_request(
 
 // client //
 
+static void log_buffer_with_rate_limiting(const char* msg, iobuf& buf) {
+    static constexpr int buffer_size = 0x100;
+    static constexpr auto rate_limit = std::chrono::seconds(1);
+    thread_local static ss::logger::rate_limit rate(rate_limit);
+    auto log_with_rate_limit = [](ss::logger::format_info fmt, auto... args) {
+        s3_log.log(ss::log_level::warn, rate, fmt, args...);
+    };
+    iobuf_istreambuf strbuf(buf);
+    strbuf.underflow();
```

Comment:
It's not obvious to me why the underflow call is needed?  (isn't strbuf going to start in the state we want it, pointing at the start of `buf`?)